tasks/main.yml

```diff
@@ -16,8 +16,31 @@
     - node_exporter_configure
     - node_exporter_run
 
+- name: Check if node_exporter is installed
+  stat:
+    path: "{{ node_exporter_destination }}"
+  register: __node_exporter_is_installed
+  tags:
+    - node_exporter_install
+
+- name: See stat (REMOVE THIS LATER)
+  debug:
+    msg: "{{ __node_exporter_is_installed }}"
+
+- name: Gather currently installed node_exporter version (if any)
+  command: "{{ node_exporter_destination }} --version"
+  args:
+    warn: false
+  changed_when: false
+  register: __node_exporter_current_version_output
+  when: __node_exporter_is_installed.stat.exists
+  tags:
+    - node_exporter_install
```

Comment:
If this belongs to `node_exporter_install` tag, then it should be in `install.yml`

Reply:
@paulfantom Any suggestions how to tag it, or remove it completely? I put it in main on purpose, so it wouldn't include/run tasks not required.

Reply:
That's exactly what `preflight.yml` is for. It is supposed to gather information about world so role can later make decisions based on this data.

Reply:
Moved checks to preflight.